content/docs/reference/design-docs/signaling/blocks-and-signals.en.md

```diff
@@ -0,0 +1,181 @@
+---
+title: "Blocks and signals"
+linkTitle: "Blocks and signals"
+weight: 20
+---
+
+
+# Blocks
+
+The blocks have several attributes:
+
+- A signaling system that corresponds to that displayed by its first signal.
+- A **path**, which is a list of direction + detector pairs (just like route paths).
+- An **entry signal**, (optional when the block starts from a buffer stop).
+- **Intermediate signals**, if any (only used by systems with distant signals).
+- An **exit signal**, (optional when the block ends at a buffer stop).
+
+The path is expressed from detector to detector so that it can be overlayed with the route graph.
+
+A few remarks:
+
+- There can be multiple blocks with the same path, as long as they have different signaling systems. Trains only use a block at a time, and ignore others.
+- Blocks do not have a state: one can rely on the dynamic state of the zones that make it up.
+- Blocks are used to figure out which signals protect which zones in a given context.
+
+## Dependencies
+
+   - route graph. For each route:
+       - `waypoints: List<DiDetector>`
+       - `signals: OrderedMap<Position, UnloadedSignal>`
+       - `speed_limits: RangeMap<Position, SpeedLimit>`, including the logic for train category limits
+   - signaling systems
+   - drivers
+
+
+# Signals
+
+Physical signal are made up of one or more logical signals, which are displayed as a single unit on the field. During simulation, logical signals are treated as separate signals.
+
+Each logical signal is associated with a signaling system, which defines if the
+signal transmits Movement Authority, speed limits, or both.
+
+Logical signals have one or more drivers. Signal drivers are responsible for computing
+signal state. Any given signal driver only works for a given pair of signaling systems,
+where the first one is displayed by the signal, and the second is the one displayed by
+the next signal.
+
+When a logical signal has an empty driver list, its content is deduced from neighboring signals.
+
+For example, a BAL signal that is both a departure of the TVM block and a
+departure of the BAL block, it will have two drivers: `BAL-BAL` and `BAL-TVM`.
+
+## Announcing speed limits
+
+When a signal announces a speed limit, it needs to be linked with a speed section object.
+This is meant to enable smooth transitions between the reaction to the announce signal, and the limit itself.
+
+If multiple signals are involved in the announce process, only the one closest to the speed limit has to have
+this attribute set.
+
+```yaml
+{
+    # ...
+    "announce_speed_section": "${SPEED_SECTION_ID}"
```

Comment:
We do not really understand what this allows. 
How can we specify that certain routes have no speed sections to announce ?
Is the purpose of this field only to reduce the number of conditional parameters ?